pkg/docker/docker.go

```diff
@@ -241,6 +242,65 @@ func (c *Client) SupportedPlatforms(ctx context.Context) ([]v1.Platform, error)
 	}, nil
 }
 
+// ImageDistribution fetches the details for the specified image by asking
+// docker to fetch the distribution manifest from the registry. The digest
+// retrieved may not appear accurate when compared to the hub website but
+// this is expected.
+//
+// cf:
+//   - https://github.com/moby/moby/issues/40636)
+//   - https://github.com/docker/roadmap/issues/262
+//
+// When a docker image is available on only a single platform, the digest
+// shown will be the digest pointing directly at the manifest for that image
+// on that platform (as shown by the docker hub).  Where multiple platforms
+// are available, the digest is pointing to a top level document describing
+// all of the different platform manifests available.
+//
+// In either case, `docker pull` will do the correct thing and download the
+// image for your platform. For example:
+//
+// $ docker manifest inspect  bitnami/rabbitmq@sha256:0be0d2a2 ...
+//
+//	"manifests": [ {
+//		  "digest": "sha256:959a02013e8ab5538167f9....",
+//		  "platform": { "architecture": "amd64", "os": "linux" }
+//		},
+//		{
+//		  "digest": "sha256:11ee2c7e9e69e3a8311a19....",
+//		  "platform": { "architecture": "arm64", "os": "linux"}
+//		}]
+//
+// $ docker pull bitnami/rabbitmq@sha256:0be0d2a2 ...
+// $ docker image ls
+// bitnami/rabbitmq ... 48603925e10c
+//
+// The digest 486039 can be found in manifest sha256:11ee2c7e which is the manifest for
+// the current authors machine.
+//
+// $ docker manifest inspect bitnami/rabbitmq@sha256:11ee2c7e
+//
+//	  "config": {
+//		   "size": 7383,
+//		   "digest": "sha256:48603925e10c01936ea4258f...."
+//	  }
+//
+// This is the image that will finally be installed.
+func (c *Client) ImageDistribution(ctx context.Context, image string, dockerCreds config.DockerCredentials) (*ImageManifest, error) {
+	authToken := getAuthToken(ctx, image, dockerCreds)
+
+	dist, err := c.DistributionInspect(ctx, image, authToken)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
There is some local changes @olgibbons has which also adds checking the local image cache at this step – probably worth doing if the image is not in our (memory) cache.